File: weknot_android/app/src/main/java/com/example/weknot_android/view/fragment/SocialFragment.java
package com.example.weknot_android.view.fragment;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.example.weknot_android.R;
import com.example.weknot_android.base.BaseFragment;
import com.example.weknot_android.databinding.SocialFragmentBinding;
import com.example.weknot_android.model.entity.user.Friend;
import com.example.weknot_android.viewmodel.SocialViewModel;
import com.example.weknot_android.widget.recyclerview.adapter.SocialAdapter;

public class SocialFragment extends BaseFragment<SocialFragmentBinding> {

    private SocialAdapter receiveAdapter;
    private SocialAdapter friendAdapter;

    private SocialViewModel socialViewModel;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        initViewModel();
        initData();

        observeSocialViewModel();
    }

    private void initViewModel() {
        socialViewModel = ViewModelProviders.of(this).get(SocialViewModel.class);
    }

    private void initData() {
        socialViewModel.getFriends();
    }

    private void observeSocialViewModel() {
        socialViewModel.getData().observe(this, friends -> {
            for (Friend friend: friends) {
                if (friend.getFriendPoint() == 1) {
                    socialViewModel.receiveList.getValue().add(friend);
                }
                else if (friend.getFriendPoint() == 2) {
                    socialViewModel.friendList.getValue().add(friend);
                }
            }
            receiveAdapter = new SocialAdapter(getContext(), socialViewModel.receiveList.getValue());
            friendAdapter = new SocialAdapter(getContext(), socialViewModel.friendList.getValue());

            setRecyclerView();
        });
    }

    private void setRecyclerView() {
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext());
        binding.receiveRecyclerview.setAdapter(receiveAdapter);
        binding.friendRecyclerview.setAdapter(friendAdapter);
        binding.receiveRecyclerview.setLayoutManager(linearLayoutManager);
        binding.friendRecyclerview.setLayoutManager(linearLayoutManager);
    }

    @Override
    protected int layoutId() {
        return R.layout.social_fragment;
    }
}

File: weknot_android/app/src/main/java/com/example/weknot_android/view/fragment/SocialFragment.kt
package com.example.weknot_android.view.fragment

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.lifecycle.Observer
import com.example.weknot_android.BR
import com.example.weknot_android.R
import com.example.weknot_android.base.fragment.BaseFragment
import com.example.weknot_android.databinding.SocialFragmentBinding
import com.example.weknot_android.view.activity.ProfileActivity
import com.example.weknot_android.viewmodel.SocialViewModel

class SocialFragment : BaseFragment<SocialFragmentBinding, SocialViewModel>() {

    override fun getLayoutId(): Int {
        return R.layout.social_fragment
    }

    override fun getViewModel(): Class<SocialViewModel> {
        return SocialViewModel::class.java
    }

    override fun getBindingVariable(): Int {
        return BR.viewModel
    }

    override fun initObserver() {
        with(viewModel) {
            onErrorEvent.observe(this@SocialFragment, Observer {
                simpleToast(it.message)
            })

            with(receiveAdapter) {
                checkFriendEvent.observe(this@SocialFragment, Observer {
                    friendRequest.value = it
                    putFriend()
                })

                openProfile.observe(this@SocialFragment, Observer {
                    val intent = Intent(context, ProfileActivity::class.java)
                    intent.putExtra("id", it)
                    startActivity(intent)
                })
            }

            friendAdapter.openProfile.observe(this@SocialFragment, Observer {
                val intent = Intent(context, ProfileActivity::class.java)
                intent.putExtra("id", it)
                startActivity(intent)
            })
        }
    }

    override fun onResume() {
        super.onResume()
        viewModel.setUp()
    }
}